### ISSUE TITLE
Tree: Integrate with tui-rs-tree-wdget

### ISSUE THREAD

**Issue by dzfrias** [ratatui-org/ratatui#116]
## Problem
With this crate's recent release, I think it would be nice to have some more general-purpose widgets added to the current repertoire. [tui-rs-tree-widget](https://github.com/EdJoPaTo/tui-rs-tree-widget) is one such widget that I think this crate would benefit from having. As discussed in [this issue](https://github.com/EdJoPaTo/tui-rs-tree-widget/issues/18), both the maintainer and I are interested in this merge.

## Solution
Add the `Tree` widget from `tui-rs-tree-widget` into this repository.

**Comment by mindoodoo** [comment]
Ay, I've actually used `tui-tree-widget` before ! It's awesome, worked like a charm !

Since we recently release our first version of ratatui, we are still in the process of evaluating where we want to go next.  With that in mind, we have created a [discussion](https://github.com/tui-rs-revival/ratatui/discussions/96) and are currently in the process of planning a meeting to discuss this further.

If you're able to make it, we would love to have you join in order to discuss this issue further. We have a poll running to figure out the time slot [here](https://calendly.com/d/y2h-b3c-n7s/ratatui-3) and you can find more information on our [discord.](https://discord.gg/pMCEU9hNEj)

All in all, I'm personally in favor of adding this in, we just need to make sure we do it in a coherent manner.

**Comment by mindoodoo** [comment]
Could you create a PR for this ? We're still figuring stuff out organization wise (hence the long follow up time), but there's interest for this widget and we are now working on merging another external widget so now is the good time ! 🚀

**Comment by dzfrias** [comment]
Okay, I can work on this when I have more time cleared up. Alternatively, @EdJoPaTo (the maintainer) could work on this if they have the time.

**Comment by mindoodoo** [comment]
Alright sounds good !

**Comment by joshka** [comment]
After giving this a lot of thought, I think it would be better to keep this out of the core Ratatui library. I've just started a new `ratatui-widgets` crate that will have a lot of widgets, with a goal of consistent UI, event handling etc. The treeview is a goal there, and I think there's a lot of this widget that would influence that one. Once we've worked out the common ways to display widgets and handle user interaction, I'd definitely want to start working on the tree view there instead of here.

Does this make sense?
